spec/lib/post_revisor_spec.rb

```diff
@@ -422,6 +422,20 @@
         }.to change { post.topic.bumped_at }
       end
 
+      it "does not bump topic when no topic category" do
+        topic_with_no_category = Fabricate(:topic, category_id: nil)
+        post_from_topic_with_no_category = Fabricate(:post, topic: topic_with_no_category)
+        expect {
+          result =
+            subject.revise!(
+              Fabricate(:admin),
+              raw: post_from_topic_with_no_category.raw,
+              tags: ["foo"],
+            )
+          expect(result).to eq(true)
+        }.to change { topic.reload.bumped_at }
```

Comment:
The test description says "not bump topic"... but here we're checking that it was bumped 🤔

Reply:
fiddling with topic tracking state feels risky. not even sure what a topic without a category is? even uncategorized/general gets a category id.

Reply:
Yeah, a topic without a category should not exist. But our lack of enforced foreign-keys makes it possible, so we need to be able to handle the situation without blowing up.